examples/nuxt-3/composables/checkout/index.ts

```diff
@@ -0,0 +1,66 @@
+import { get, set, del } from 'idb-keyval';
+import createShopifyCheckoutClient, {
+  CheckoutClient
+} from '@nacelle/shopify-checkout';
+import { ShopifyCheckout } from '@nacelle/shopify-checkout/dist/types/checkout-client.types';
+import createCheckout from '@nacelle/shopify-checkout/dist/types/client/actions/checkoutCreate';
+import { Ref } from 'vue';
+const checkoutClient: Ref<CheckoutClient | null> = ref(null);
+
+const checkoutState: { checkoutId?: string } = reactive({
+  checkoutId: null
+});
+
+export function useShopifyCheckout(): CheckoutClient {
+  if (!checkoutClient.value) {
+    const runTimeConfig = useRuntimeConfig();
+    checkoutClient.value = createShopifyCheckoutClient({
+      storefrontCheckoutToken:
+        runTimeConfig.public.shopify.StorefrontCheckoutToken,
+      myshopifyDomain: runTimeConfig.public.shopify.myShopifyDomain
+    });
+  }
+  return reactive(checkoutClient.value);
+}
+
+export async function initCheckout(): Promise<void> {
+  try {
+    const shopifyCheckoutClient = useShopifyCheckout();
+    checkoutState.checkoutId = await get('checkoutId');
+    if (checkoutState.checkoutId) {
+      const checkout = await shopifyCheckoutClient.get({
+        id: checkoutState.checkoutId
+      });
+      if (checkout?.completed) {
+        await del('checkoutId');
+        await clearCart();
+      }
+    }
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+export async function processCheckout(): Promise<void> {
+  try {
+    const cartState = useCart();
+    const shopifyCheckoutClient = useShopifyCheckout();
+    const checkoutItems = cartState.lineItems.map((lineItem) => ({
+      quantity: lineItem.quantity,
+      variantId: lineItem.variant.id
+    }));
+    const checkoutArgs = { cartItems: checkoutItems };
```

Comment:
It looks like `checkoutArgs` needs a type. We could use an inline type here:
```suggestion
    const checkoutArgs: { cartItems: typeof checkoutItems; id?: string } = {
      cartItems: checkoutItems
    };
```

Reply:
This looks like TS is able to generate an anonymous type from the constructed objects. 
<img width="500" alt="VS Code typescript popup showing checkoutArgs having a type of an array of objects with a cartItems key. That cartItems key has a type of objects that contain quantity: number and variantId: string" src="https://user-images.githubusercontent.com/30670444/185507774-74e826ca-a9fb-4713-82d7-63191802cddf.png">